ps3.py

```diff
@@ -0,0 +1,414 @@
+    ceil(n/3) letters in the hand should be VOWELS (note,
+    ceil(n/3) means the smallest integer not less than n/3).
+
+    Hands are represented as dictionaries. The keys are
+    letters and the values are the number of times the
+    particular letter is repeated in that hand.
+
+    n: int >= 0
+    returns: dictionary (string -> int)
+    """
+
+    hand = {}
+    num_vowels = int(math.ceil(n / 3) - 1)
+    wildcard = {'*': 1}
+
+    for i in range(num_vowels):
+        x = random.choice(VOWELS)
+        hand[x] = hand.get(x, 0) + 1
+
+    for i in range(num_vowels, n):
+        x = random.choice(CONSONANTS)
+        hand[x] = hand.get(x, 0) + 1
+
+    hand.update(wildcard)
+    return hand
+
+
+def update_hand(hand, word):
+    """
+    Does NOT assume that hand contains every letter in word at least as
+    many times as the letter appears in word. Letters in word that don't
+    appear in hand should be ignored. Letters that appear in word more times
+    than in hand should never result in a negative count; instead, set the
+    count in the returned hand to 0 (or remove the letter from the
+    dictionary, depending on how your code is structured). 
+
+    Updates the hand: uses up the letters in the given word
+    and returns the new hand, without those letters in it.
+
+    Has no side effects: does not modify hand.
+
+    word: string
+    hand: dictionary (string -> int)    
+    returns: dictionary (string -> int)
+    """
+    new_hand = hand.copy()
+
+    for i in word.lower():
+        if i in new_hand:
+            amount = new_hand.get(i)
+            el = {i: amount - 1}
+            new_hand.update(el)
+
+    for i in hand:
+        if new_hand[i] == 0:
+            new_hand.pop(i)
+
+    return new_hand
+
+
+def is_valid_word(word, hand, word_list):
+    """
+    Returns True if word is in the word_list and is entirely
+    composed of letters in the hand. Otherwise, returns False.
+    Does not mutate hand or word_list.
+   
+    word: string
+    hand: dictionary (string -> int)
+    word_list: list of lowercase strings
+    returns: boolean
+    """
+    word = word.lower()
+    asterix = '*'
+
+    for j in VOWELS:
+        normal_word = word.replace(asterix, j)
+        if normal_word in word_list:
+            for i in set(word):
+                if i not in hand or word.count(i) > hand[i]:
+                    return False
+            return True
+
+    return False
+
+
+def calculate_handlen(hand):
+    """ 
+    Returns the length (number of letters) in the current hand.
+    
+    hand: dictionary (string-> int)
+    returns: integer
+    """
+    letter_amount = len(hand)
+    return letter_amount
+
+
+def ask_user(*args):
+    """
+    message: message for input
+    pattern: pattern for a variable
+    returns: value of variable
+    """
+    input_text = input(args[0])
+    while not re.fullmatch(args[1], input_text):
+        input_text = input(args[0])
```

Comment:
Could you also add ability to show a warning message, where you will be able to remind user which inputs are acceptable?